doc/src/manual/performance-tips.md

```diff
@@ -1631,3 +1631,21 @@ will not require this degree of programmer annotation to attain performance.
 In the mean time, some user-contributed packages like
 [FastClosures](https://github.com/c42f/FastClosures.jl) automate the
 insertion of `let` statements as in `abmult3`.
+
+## [Multithreading and linear algebra](@id man-multithreading-linear-algebra)
+
+This section applies to multithreaded Julia code which, in each thread, performs linear algebra operations.
+Indeed, these linear algebra operations involve BLAS / LAPACK calls, which are themselves multithreaded.
+In this case, one must ensure that cores aren't oversubscribed due to the two different types of multithreading.
+
+Julia compiles and uses its own copy of OpenBLAS for linear algebra, whose number of threads is controlled by the environment variable `OPENBLAS_NUM_THREADS`.
+It can either be set as a command line option when launching Julia, or modified during the Julia session with `BLAS.set_num_threads(N)` (the submodule `BLAS` is exported by `using LinearAlgebra`).
+Its current value can be accessed with `BLAS.get_num_threads()`.
+When the user does not specify anything, the default setting is `OPENBLAS_NUM_THREADS=8` (for Julia version < 1.8) or `OPENBLAS_NUM_THREADS=Sys.CPU_THREADS` (for Julia version ≥ 1.8).
```

Comment:
Mentioning the default behavior here is a bit "risky" since the behavior has changed a bunch in the recent past and someone would need to make sure that this is always correct/up to date. I've written the cited text quite some time ago and I'm not even sure that Julia just uses `Sys.CPU_THREADS` as default in 1.9 or later. (Also, at some point I've wondered whether we should even vary the default for `OPENBLAS_NUM_THREADS` based on whether Julia is run in single or multi-threaded mode. After all, the current setting is a footgun in the latter case.)

Anyways, maybe we can/should make a looser statement. Something like this:

"By default, that is, when the user does not specify anything, Julia tries to choose a reasonable value for the number of BLAS threads, e.g. `OPENBLAS_NUM_THREADS=Sys.CPU_THREADS`. However, setting the value manually is typically superior."

Reply:
(Also, at some point I've wondered whether we should even vary the default for `OPENBLAS_NUM_THREADS` based on whether Julia is run in single or multi-threaded mode. After all, the current setting is a footgun in the latter case.)

Reply:
> (Also, at some point I've wondered whether we should even vary the default for `OPENBLAS_NUM_THREADS` based on whether Julia is run in single or multi-threaded mode. After all, the current setting is a footgun in the latter case.)

Well, the first thing is documenting the footgun, then we'll think about solving it :)